src/main/java/build/buildfarm/worker/SuperscalarPipelineStage.java

```diff
@@ -14,22 +14,16 @@
 
 package build.buildfarm.worker;
 
-import java.util.concurrent.ArrayBlockingQueue;
 import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.Semaphore;
 import java.util.concurrent.TimeUnit;
-import java.util.logging.Level;
 
 abstract class SuperscalarPipelineStage extends PipelineStage {
-  protected final int width;
-
   @SuppressWarnings("rawtypes")
-  protected final BlockingQueue claims;
+  protected Slots slots;
```

Comment:
Make this final, and would prefer initialization with width parameterization - don't use class just for containerization.

Reply:
fixed